core/crop/structure.py

```diff
@@ -97,35 +101,41 @@ class SensorClass(BASE):
     airvelocity_readings_relationship = relationship("ReadingsAirVelocityClass")
     environmental_readings_relationship = relationship("ReadingsEnvironmentalClass")
 
+    # relationshionships (Many-To-One)
+    #location_relationship = relationship("LocationClass")
+
     # arguments
     __table_args__ = (UniqueConstraint("type_id", "device_id", name="_type_device_uc"),)
 
 
 class LocationClass(BASE):
     """
     This class describes all the physical locations in the farm. eg. Sensor x is found in
-    the front section, in the left column , in the 3rd self.
+    the zone Farm 1, in the B aisle, In column 23 in the 4rd self.
+        Description of location codes: 
+        zones: (String) description of zone eg. Entrance, stairs, Farm 1, etc.. 
+        aisle: (String) A, B. (Enumeration of aisles)
+        column: (Integer). Enumeration of columns in farm zones.
+        shelf: (Integer) Number of shelves per column from bottom to top: 1-4. 
     """
 
     __tablename__ = LOCATION_TABLE_NAME
 
     # columns
-    id = Column(Integer, primary_key=True, autoincrement=True)
+    id = Column(Integer, primary_key=True)
 
-    zone = Column(String(50), nullable=False)  # farm zone
-    aisle = Column(String(50), nullable=False)  # Section A/B
-    column = Column(Integer, nullable=False)  # no
-    shelf = Column(Integer, nullable=False)  # 1,2,3,4 / top/middle/bottom/?
-    code = Column(String, nullable=False, unique=True)
-
-    time_created = Column(DateTime(), server_default=func.now())
-    time_updated = Column(DateTime(), onupdate=func.now())
+    zone = Column(String(50), nullable=False)
+    aisle = Column(String(50), nullable=False)
+    column = Column(Integer, nullable=False)
+    shelf = Column(Integer, nullable=False)
+    #FIXME: the following is not working with orm
+    #code = column_property(section + column + shelf) #generated code of location
```

Comment:
Why do we change this?

Reply:
just removed the comments and added them in the beginning of the function

Reply:
I should have been more specific here - are we sure that we are not going to use code anymore? If so then you need to take care of other instance that it was previously used, e.g. 
    # constructor
```
    def __init__(self, zone, aisle, column, shelf, code):
        self.zone = zone
        self.aisle = aisle
        self.column = column
        self.shelf = shelf
        self.code = code
```